src/engine/universal/ui/src/display-items/tasklist/TaskList-DisplayItem.js

```diff
@@ -70,8 +70,22 @@ class TaskListTab extends DisplayItem {
       }
 
       definitionId = engine.definitionId;
-      variables = userTask.processInstance.getVariables(userTask.tokenId);
-      milestonesData = engine.getMilestones(query.instanceID, query.userTaskID);
+
+      if (userTask.state === 'READY' || userTask.state === 'ACTIVE') {
+        variables = userTask.processInstance.getVariables(userTask.tokenId);
+        milestonesData = engine.getMilestones(query.instanceID, query.userTaskID);
+      } else {
+        // get info of non-active userTask from logs
+        const logs = userTask.processInstance.getState().log;
+        const userTaskLogEntry = logs.find(
+          (logEntry) =>
+            logEntry.flowElementId === userTask.id &&
+            logEntry.tokenId === userTask.tokenId &&
+            logEntry.startTime === userTask.startTime
+        );
+        variables = userTask.processInstance.getVariables();
```

Comment:
Wouldn't this show the value of the variables that they have in the current instance state instead of the values the user task was submitted with? Is this the desired behaviour?